src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -192,23 +236,81 @@ public void setNumberOfInstances(int numberOfInstances) {
 
 			List<String> urls = sas.getUrls();
 			if (urls != null && !urls.isEmpty()) {
-				v.setAccessURL(this.determineAccessURL(v.getTarget().getProtocol(), urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns(), v.getTarget().getPath()));
+				// Set the access url to the selected route (without any protocol or path yet)
+				v.setAccessURL(this.determineApplicationRoute(urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));				
+      }      
+           
+			v.setNumberOfInstances(sas.getInstances());
+			
+			return Mono.just(v);
+    });
+        
+    Flux<List<RouteResource>> routeApplicationFlux = osaVectorSpaceFlux.flatMapSequential(v -> this.getAppRoutes(v.getApplicationId()));
+		Flux<OSAVector> osaVectorApplicationRouteFlux = Flux.zip(osaVectorApplicationFlux, routeApplicationFlux).flatMap(tuple -> {
+			OSAVector v = tuple.getT1();
+			
+			if (INVALID_ROUTES == tuple.getT2()) {
+				// NB: This drops the current target!
+				return Mono.empty();
 			}
 			
-			v.setNumberOfInstances(sas.getInstances());
+      List<RouteResource> routes = tuple.getT2();
+            
+      RouteResource metricsRoute = routes.stream()
+      .filter(r -> v.getAccessURL().contains(r.getUrl()))
+      .findFirst()
+			.orElse(null);
+
+			if (metricsRoute == null) {
+				return Mono.just(v);
+			}
+
+			Destination dest = metricsRoute.getDestinations()
+			.stream()
+			.filter(d -> d.getApplication().getApplicationId().equals(v.getApplicationId()))
```

Comment:
After several minutes of thinking, I finally understood why you need to do this here. However, that tells me that it might be worth a comment why you do it (one route still can have multiple destinations - and we need to find the right destination...)

Reply:
Added a clarifying comment